pkg/tuner/tuner.go

```diff
@@ -0,0 +1,95 @@
+package tuner
+
+import (
+	"fmt"
+	"net/url"
+	"os/exec"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/vshn/waf-tool/cfg"
+	"github.com/vshn/waf-tool/pkg/elasticsearch"
+	"github.com/vshn/waf-tool/pkg/forwarder"
+	"github.com/vshn/waf-tool/pkg/model"
+	"github.com/vshn/waf-tool/pkg/rules"
+)
+
+const baseID = 10100
+
+// Tune creates exclusion rules for a given uniqe ID
+func Tune(uniqueID string, config cfg.Configuration) (returnError error) {
+	out, err := exec.Command("oc", "whoami", "--show-token").Output()
+	if err != nil {
+		return fmt.Errorf("could not get token: %w", err)
+	}
+	token := strings.TrimSpace(string(out))
+	es, err := elasticsearch.New(config.ElasticSearch, token)
+	if err != nil {
+		return fmt.Errorf("could not create ES client: %w", err)
+	}
+
+	url, err := url.Parse(config.ElasticSearch.URL)
+	if err != nil {
+		return err
+	}
+
+	fwd := forwarder.New(url.Port())
+
+	defer func() {
+		if err := fwd.Stop(); err != nil {
+			if returnError == nil {
+				returnError = err
+			} else {
+				// Already encountered an error so we log this one here
+				log.WithError(err).Error("Could not stop port forwarding")
+			}
+		}
+	}()
+	if err := fwd.Start(); err != nil {
+		return err
+	}
+
+	log.WithFields(log.Fields{
+		"url": config.ElasticSearch.URL,
+	}).Debug("Connecting to Elasticsearch...")
```

Comment:
This is basically just setting up the ES client. It would make sense to create a private function called "prepareEsClient" or similar, for more code readability. Half of the method is just understanding that. With moving to a more English-named function we avoid commenting what this chunk does (Clean code) and focus on the business logic

Reply:
Also, `oc` assumes that the binary is called like that and somewhere in PATH. It should also be configurable in the future, but we can leave it like that for the time being.